### PR TITLE
[CBRD-25358] error handling when using for update clause for system tables such as db_root, dual, etc.

### PR DESCRIPTION
**Pull request by beyondykk9**
http://jira.cubrid.org/browse/CBRD-25358

No errors occur when using the "for update" clause for system tables such as db_root and dual. Semantic error handling is required.

`select /*+ recompile */ * from db_root for update; `

expected result:
ERROR: update is not autorized on db_root.

### CODE REVIEW

File: src/parser/name_resolution.c [hunk 1]
```diff
@@ -9198,6 +9216,17 @@ pt_resolve_names (PARSER_CONTEXT * parser, PT_NODE * statement, SEMANTIC_CHK_INF
...
+	      for (entity = spec->info.spec.flat_entity_list; entity; entity = entity->next)
+		{
+		  db = entity->info.name.db_object;
+		  if (db_is_system_class (db) && (db_check_authorization (db, DB_AUTH_UPDATE) != NO_ERROR))
+		    {
+		      PT_ERRORmf2 (parser, entity, MSGCAT_SET_PARSER_RUNTIME, MSGCAT_RUNTIME_IS_NOT_AUTHORIZED_ON,
+				   "UPDATE", entity->info.name.original);
+		      return NULL;
+		    }
+		  spec->info.spec.flag = (PT_SPEC_FLAG) (spec->info.spec.flag | PT_SPEC_FLAG_FOR_UPDATE_CLAUSE);
```
Comment (hyunikn) [inline src/parser/name_resolution.c:9228]: This is done already at line 9218.

File: src/parser/name_resolution.c [hunk 2]
```diff
@@ -9187,6 +9194,17 @@ pt_resolve_names (PARSER_CONTEXT * parser, PT_NODE * statement, SEMANTIC_CHK_INF
 			      node->info.name.original);
 		  return NULL;
 		}
+
+	      for (entity = spec->info.spec.flat_entity_list; entity; entity = entity->next)
+		{
+		  db = entity->info.name.db_object;
+		  if (db_is_system_class (db) && (db_check_authorization (db, DB_AUTH_UPDATE) != NO_ERROR))
```
Comment (ctshim) [inline src/parser/name_resolution.c:9201]: There is the same code on line 9222.
Do both conditions have to be met? Isn't this an OR condition?

Comment (beyondykk9) [reply src/parser/name_resolution.c:9201]: This routine is for "FOR UPDATE OF class-name-list" and the below routine is for "FOR UPDATE" without "OF" clause.

File: src/parser/name_resolution.c (hunk 1 above)
Comment (beyondykk9) [reply src/parser/name_resolution.c:9228]: This routine is for "FOR UPDATE" without "OF" clause and the above routine is for "FOR UPDATE OF class-name-list" clause.

**Review by hgryoo — APPROVED** [review]

**Review by hyunikn — APPROVED** [review]

**Review by jongmin-won — APPROVED** [review]

**Review by ctshim — APPROVED** [review]

**Pull request merged**